### DISCUSSION TITLE
low level touch coords seem to be fixed in portrait mode

### DISCUSSION THREAD

**Discussion by DougK71** [rzeldent/esp32-smartdisplay#93] [General]
I designed a display in SLS 800w 480h labels buttons sliders are all placed in SLS with the SLS placement view in landscape. When compiles the buttons sliders etc only work in portrait mode 
I appears the landscape position of a widget that interfaces with the low level touch gets portrait coords so that clicking on the landscape position in (800w 480h) coords does not correspond to touch coords of 480w 800h so the low level code that determines if a touch is in the zone of a widget can not trigger the call back

**Comment by DougK71** [comment]
the display is esp32-8048s070c and the touch controller is gt911

**Comment by DougK71** [comment]
I don't know how to calibrate a landscape touch so I am using the demo screen which is in portrait mode to get initial set up
The my display design using SLS looks good and as a test I can rotate the graphics ( I don't need it as the design was done in landscape mode) and did it to see if the touch to display co-ordinate transform rotated the touch
Well touch works since it seems the touch calibration data array never changes and it is buried some place I haven't found like compiled in the touch support code
I need help to run either a new touch calibration or a call to something like swap x for y and y for x

**Comment by rzeldent** [comment]
Hi Dough,

There seem to be multiple versions of the esp32-8048s070c. See issue: https://github.com/rzeldent/esp32-smartdisplay/issues/90. Right now I have a 1.1 and 1.3 revision. See https://github.com/rzeldent/platformio-espressif32-sunton/tree/develop.

Maybe there are also differences in the touch? If you want to experiment, toy can change the SWAP in the esp32-8048s070c board json file; this can be found in the directory boards.

Let me know if this helps and what works!

**Comment by DougK71** [comment]
Thank you for the advice
It seems I was blindsided by the necessary calibration a resistive touch needs
Capacative touch likely has a fixed coord system and the necessity is to map from it to the screen orientation ...a bit like the metric tensor in General Relativity. 
Now if using your portrait template and I rotate from landscape to portrait using 
**lv_disp_set_rotation(disp, LV_DISP_ROT_90);** the buttons still visible all work just fine and using
**horz_res=lv_disp_get_hor_res(disp);
   vert_res=lv_disp_get_ver_res(disp);**
I see hor=480 and ver=800
After removing the set rotation command the display graphics are correct  in landscape mode 
I see hor=800 and ver=480 so the display knows its coords
Now a left action on the touch activates a right button and an up touch actives a down button and vice versa  as if the touch code is still mapping to the hor=480 vert=800 display coords
There is code in the gt911 that expects to be made aware of the display coord system and it looks like the gt911 performs the coord translation and returns 3 screen tuples screen[3] and 3 tp tuples touch[3]
I suspect your have the call to the gt911 driver always with 480w 800 h display coords and that a call to rotate isn't calling the gt911 driver to remap the mapping only the display is remapped. 
gt911 has 4 mapping choices as well as min X max X min Y maxY perhaps giving manufacturers a choice of how they stick the capacative glass to the LCD ( back to front and/or rotate 180 )
I don't see where SWAP fits in but at this point I'll try everything to get this to work
Note your demo code unmodified works fine so I don't suspect it is the LCD interface but I do suspect the touch interface

Help I don't know how to do something like below to see the screen to touch mapping in the terminal
**Serial.println(touch_calibration_data_t);
 Serial.println(lv_point_t touch);**
}

**Comment by DougK71** [comment]
You guessed right....Oops I misunderstood what you referred to as SWAP since it often refers to bit order
Anyway by changing **"-DTOUCH_SWAP_X=false", "-DTOUCH_SWAP_Y=false",** from true touch and widgets using touch are now working correctly. Often you learn more when things don't work than when they do.
Thank you
